include/osquery/events.h

```diff
@@ -448,6 +448,28 @@ class EventSubscriberPlugin : public Plugin {
    */
   Status recordEvent(EventID& eid, EventTime time);
 
+  /**
+   * @brief Get the expiration timeout for this event type
+   *
+   * The default implementation retrieves this value from FLAGS_events_expiry.
+   * This method can be overridden to allow custom event expiration timeouts in
+   * subclasses of EventSubscriberPlugin.
+   *
+   * @return The events expiration timeout for this event type
+   */
+  virtual uint64_t getEventsExpiry();
```

Comment:
In regard to `size_t`, I don't think there's any harm in it, but `uint64_t` is consistent with the definition of the flags (https://github.com/facebook/osquery/blob/master/osquery/events/events.cpp#L39). I guess `size_t` should be equivalent to `uint64_t` everywhere these days?